Sapphire network changes

grvgoel81 opened the pull request:
* Added sapphire network changes.
* Updated and modified unit-tests.

<img width="856" alt="Screenshot 2024-09-02 at 4 16 50 PM" src="https://github.com/user-attachments/assets/dee33dc0-a5ae-45ac-acbd-cbb45e7925a5">

grvgoel81 committed:
feat: update torus-utils-java sdk, modified code to support sapphire network

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: update and added unit-tests

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
Merge branch 'master' into sapphire-network

# Conflicts:
#	singlefactorauth/build.gradle

grvgoel81 committed:
feat: update torus-utils.java version and code updated wrt new sdk

Signed-off-by: Gaurav Goel <gaurav@tor.us>

shahbaz17 commented on app/src/main/java/com/web3auth/sfaexample/MainActivity.kt:
```diff
@@ -38,12 +38,14 @@ class MainActivity : AppCompatActivity() {
-        val sessionResponse: CompletableFuture<TorusKey> = singleFactorAuth.initialize(this.applicationContext)
-        sessionResponse.whenComplete { torusKey, error ->
+        sfaParams =
+            SFAParams(Web3AuthNetwork.SAPPHIRE_MAINNET, "YOUR_CLIENT_ID", false)
```
which `false` here? what does it do?

Does SFAParams take three input now?

shahbaz17 replied:
Got it. This is for usePnPKey.

shahbaz17 commented on singlefactorauth/build.gradle:
```diff
@@ -37,7 +37,7 @@ dependencies {
 
     implementation 'org.web3j:crypto:4.8.9-android'
-    implementation 'org.torusresearch:torus-utils-java:3.4.0'
+    implementation 'com.github.grvgoel81:torus-utils-java:7.0.0'
```
why this repo and not web3auth's?

grvgoel81 replied:
Its not final since torus-utils-java is under review.

metalurgical replied:
We should get that merged in first and then update it in this PR.

shahbaz17 commented on singlefactorauth/src/test/java/com/web3auth/singlefactorauth/AquaTest.kt:
```diff
@@ -32,8 +31,8 @@ class AquaTest {
     fun shouldGetTorusKey() {
-        singleFactorAuthArgs = SingleFactorAuthArgs(Web3AuthNetwork.AQUA, "YOUR_CLIENT_ID")
-        singleFactorAuth = SingleFactorAuth(singleFactorAuthArgs)
+        sfaParams = SFAParams(Web3AuthNetwork.AQUA, "YOUR_CLIENT_ID")
```
Test doesn't use `usePnPKey`

metalurgical replied:
usePnPKey variable was removed.

grvgoel81 committed:
feat: PR comments addressed

Signed-off-by: Gaurav Goel <gaurav@tor.us>

metalurgical commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt:
```diff
@@ -9,37 +9,38 @@ import com.web3auth.singlefactorauth.types.AggregateVerifierParams
-            SingleFactorAuthArgs.SIGNER_MAP[singleFactorAuthArgs.getNetwork()] + "/api/sign"
-        opts.allowHost =
-            SingleFactorAuthArgs.SIGNER_MAP[singleFactorAuthArgs.getNetwork()] + "/api/allow"
+        val opts = TorusOptions(
```
You're missing a parameter here from singlefactorauthargs, because SFAParams is in turn missing  a parameter in it's constructor.

metalurgical commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt:
```diff
@@ -49,26 +50,24 @@ class SingleFactorAuth(singleFactorAuthArgs: SingleFactorAuthArgs) {
-    ): CompletableFuture<TorusKey> {
-        val torusKeyCompletableFuture: CompletableFuture<TorusKey> =
-            CompletableFuture<TorusKey>()
+    ): CompletableFuture<SFAKey> {
```
CompletableFuture<SFAKey> can be simplified to TorusSFAKey.

You're already waiting for the result so this future is unnecessary and can be simplified into:
```
val sessionResponse: String =
                sessionManager.authorizeSession(false).get()
// then process the result
```

metalurgical commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt:
```diff
@@ -122,13 +121,16 @@ class SingleFactorAuth(singleFactorAuthArgs: SingleFactorAuthArgs) {
-            torusUtils.getOrSetNonce(pubDetails.x, pubDetails.y, false).get()
+    ): TorusKey {
+        if (pubDetails.metadata.typeOfUser.equals(TypeOfUser.v1)) {
+            MetadataUtils.getOrSetNonce(
+                "", pubDetails.metadata.pubNonce.x, pubDetails.metadata.pubNonce.y,
+                0, null, false, null
```
We should not be calling MetadataUtils directly outside of TorusUtils.

Please see here for how this should be implemented:
https://github.com/Web3Auth/single-factor-auth-swift/blob/main/Sources/SingleFactorAuth/SingleFactorAuth.swift#L30-L77

grvgoel81 committed:
feat: PR comments addressed

Signed-off-by: Gaurav Goel <gaurav@tor.us>

metalurgical commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt:
```diff
@@ -49,26 +49,24 @@ class SingleFactorAuth(singleFactorAuthArgs: SingleFactorAuthArgs) {
-        val retrieveSharesResponse = getRetrieveSharesResponse(loginParams, details, pubDetails)
-        if (retrieveSharesResponse.privKey == null) {
+        val retrieveSharesResponse = getRetrieveSharesResponse(loginParams, nodeDetails, pubDetails)
+        if (retrieveSharesResponse.finalKeyData.privKey == null) {
```
This is not going to work
```
        if (retrieveSharesResponse.finalKeyData.privKey == null) {
```

See here:
https://github.com/torusresearch/torus-utils-java/pull/28/files#diff-45201af8a5356f716eac4e63a4b783a84262d3d66bb2c9de18e9b95d4930524dR12

And here:
https://github.com/torusresearch/torus-utils-java/pull/28/files#diff-b989ac57484b4c05fd81ef8d37f334f9a318f7a9ba99e4f3e825e891c3e6cbc9R13


```suggestion
        if (retrieveSharesResponse.getFinalKeyData().getPrivKey() == null) {
```

Apply to all similar.

metalurgical committed:
fixup

Realign to swift implementation.
Move tests to instrumentation tests so they are runnable.
Fix tests.

metalurgical committed:
update

metalurgical committed:
update

metalurgical committed:
Update SingleFactorAuthTest.kt

metalurgical committed:
remove RequiresApi

metalurgical committed:
Update SingleFactorAuth.kt

metalurgical committed:
Merge pull request #8 from Web3Auth/sapphire-network-revision

fixup

metalurgical committed:
update

metalurgical committed:
update torus-utils to release version

metalurgical committed:
cleanup

metalurgical committed:
use same version of web3j as torus-utils

grvgoel81 committed:
feat: update minsdk version to 26

Signed-off-by: Gaurav Goel <gaurav@tor.us>

metalurgical committed:
update

chaitanyapotti commented:
the repo and params don't match up with the web counterpart. we should make it match. https://github.com/Web3Auth/single-factor-auth-web

metalurgical committed:
fix: jitpack

metalurgical committed:
update session manager to latest release

metalurgical committed:
fix: app

metalurgical commented:
@chaitanyapotti This was matched to the current swift implementation.

grvgoel81 committed:
feat: update example and add for sessionId check

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: Rename SingleFactorAuthArgs to SFAParams

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: Rename function getKey to connect everywhere

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 commented:
> the repo and params don't match up with the web counterpart. we should make it match. https://github.com/Web3Auth/single-factor-auth-web

Added all those changes as per discussion with devrels.

metalurgical commented:
@grvgoel81 Yes, I am aware. We had to get this to the swift head first. I think lets rather raise the last three commits as a new PR into this one.

Then I will look at making the same changes to swift (or you can if you'd like) and we should get a PR for the additional changes that looks very similar between these two and are both easily reviewable together.

Thought on this?

metalurgical commented:
@chaitanyapotti snyk is failing on a vulnerability which is excluded in torus-utils, does this need to be revisited or are we going to ignore this vulnerability here as well? It's coming from the web3j dependency.

chaitanyapotti commented:
We can ignore it for now. Wait until web3j releases a fix

grvgoel81 commented:
> @grvgoel81 Yes, I am aware. We had to get this to the swift head first. I think lets rather raise the last three commits as a new PR into this one.
> 
> Then I will look at making the same changes to swift (or you can if you'd like) and we should get a PR for the additional changes that looks very similar between these two and are both easily reviewable together.
> 
> Thought on this?

Done i separate PR: https://github.com/Web3Auth/single-factor-auth-android/pull/9

grvgoel81 committed:
feat: update session-manager-android sdk version

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: update session-manager-android sdk version

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: update session-manager-android sdk version

Signed-off-by: Gaurav Goel <gaurav@tor.us>

metalurgical committed:
Merge branch 'sapphire-network' into feat/change-class-and-function-names

metalurgical committed:
Merge pull request #9 from Web3Auth/feat/change-class-and-function-names

Change class and function names as per Web SFA sdk.

grvgoel81 committed:
feat: update torus-utils-java version to 4.2.0

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: update example

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: Rename TorusSFAKey to SFAKey everywhere

Signed-off-by: Gaurav Goel <gaurav@tor.us>

metalurgical commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/types/TorusSFAKey.kt:
```diff
@@ -1,6 +1,6 @@
 package com.web3auth.singlefactorauth.types
 
-class TorusSFAKey(
+class SFAKey(
```
Rename this file to the same name as the class.

grvgoel81 committed:
feat: Rename file name  from TorusSFAKey to SFAKey everywhere

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: Resolve json variable naming error.

Signed-off-by: Gaurav Goel <gaurav@tor.us>

chaitanyapotti commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt:
```diff
@@ -1,191 +1,128 @@
 
-        val details: NodeDetails =
+    private fun getTorusNodeEndpoints(nodeDetails: NodeDetails): Array<String?> {
+        return if (this.network.toString().contains("sapphire")) {
```
no need of such logic. can always use torusNodeEndpoints

chaitanyapotti commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt:
```diff
@@ -1,191 +1,128 @@
+        }
+    }
+
+    fun isSessionIdExists(): Boolean {
```
what's the point of this?
if sessionId is expired? what will this help with?

grvgoel81 replied:
This is for checking whether sessionId exists or not.

chaitanyapotti commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt:
```diff
@@ -1,191 +1,128 @@
+            json.put("publicAddress", torusSFAKey.getPublicAddress())
+        }
+
+        sessionManager.createSession(json.toString(), 86400, ctx, ctx.packageName).get()
```
the time should be based on sessionTime variable. not hardcoded to 86400

chaitanyapotti commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/types/SFAParams.kt:
```diff
@@ -0,0 +1,45 @@
+
+
+    init {
+        this.clientid = clientid
```
why is clientid all lowercase where as other params are camelcase?

chaitanyapotti commented on singlefactorauth/src/main/java/com/web3auth/singlefactorauth/types/SFAParams.kt:
```diff
@@ -0,0 +1,45 @@
+        return networkUrl
+    }
+    fun get(): Web3AuthNetwork {
+        return network
```
what is this param for?

grvgoel81 committed:
feat: Resolved PR comments

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: Update session-manager-android and code related changes

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: added default values for sessionTime and allowedOrigin

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: remove allowedOrigin from constructors and update all tests

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: add sessionTime in SFAParams.kt

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: add sessionTime in SFAParams.kt

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: update session-manager-sdk to 2.4.0

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: Resolved PR comments

Signed-off-by: Gaurav Goel <gaurav@tor.us>

grvgoel81 committed:
feat: Resolved PR comments

Signed-off-by: Gaurav Goel <gaurav@tor.us>

chaitanyapotti reviewed (approved):
lgtm
